Introducing Townie draft post (#135)
val-town/val-town-blog: src/content/blog/introducing-townie.mdx

---
title: Introducing Townie AI
description: Introducing Townie AI
pubDate: September 25, 2024
author: Steve Krouse
---

import { Image } from "astro:assets";

> Last month, we launched a major overhaul of Townie, our AI code assistant that helps you build, edit, and deploy your Val Town projects. Val Town is already the best place to turn an idea into a deployed app. Townie makes it even better.

Reviewer: ```suggestion
Last month, we launched a major overhaul of Townie, our AI code assistant that helps you build, edit, and deploy your Val Town projects. Val Town is already the best place to turn an idea into a deployed app. Townie gets you there much faster.
```

## Townie in Action

Want to see what it's like to build with Townie? Here's a video where I use it to make an RSVP page in just a few prompts and a few minutes. We're talking working, deployed, persistent apps in under 5 minutes. No joke!

<video controls loop class="image-rounded">
  <source src="/video/TownieIntro.mp4" />
</video>

> 

Reviewer: A short GIF might be great here, just to see the basic interaction flow

> But the best way to see what Townie is like is to [just go play with it.](https://val.town/townie)

Reviewer: Could be nice to have a button here

> ## 🔥 The Community is on Fire!

Reviewer: For all of these, I'd probably:

1. Make the image smaller. We'll need to use HTML instead of markdown images
2. Link the image themselves to the post
3. Remove the link in the text. I think the image itself can be the link, maybe

Our users have been creating some super cool stuff with Townie. Here are a few highlights:

### Blog summaries

With a hand full of prompts, [Peter made a neat tool](https://x.com/peterhartree/status/1825254990169927976) to get Claude to summarize a blog post just by pasting a link.

![Blog Summaries](./introducing-townie/blog-summaries.png)

### Mood Tracking

Marcus was able to build a really nice looking [mood-tracking app](https://x.com/MarcusSandstrom/status/1830544711850823697) in no time at all.

![Mood Tracker](./introducing-townie/mood-tracker.png)

### Slack Bots

Johnnie made a [super useful Slack bot](https://x.com/destroytoday/status/1834408549549494302) for sending door code reminders.

![Slack Bot](./introducing-townie/slack-bots.png)

## The Townie Journey

If you're curious about the evolution of Townie, check out [JP's fascinating post](https://blog.val.town/blog/codegen/) about building the early iterations of Townie on Val Town itself.

## Go Build!

> Go make something with [Townie](https://val.town/townie)! We have some helpful [prompting tips in our docs](https://docs.val.town/reference/townie/), and you get stuck, reach out in [our Discord](https://discord.gg/)!

Reviewer: Would be nice to have 3-5 ideas for what folks can make here actually

Reviewer: Could also end the article on "Build with Townie" as a button/link

Reviewer: I'd also drop the exclamation points
